exist-core/src/test/java/org/exist/xquery/WindowClauseTest.java

```diff
@@ -0,0 +1,474 @@
+    }
+
+    @Test
+    public void tumblingWindowRunUp() throws RecognitionException, XPathException, TokenStreamException, QName.IllegalQNameException {
+        final String query = "xquery version \"3.1\";\n" +
+                "for tumbling window $w in $closings\n" +
+                "   start $first next $second when $first/price < $second/price\n" +
+                "   end $last next $beyond when $last/price > $beyond/price\n" +
+                "return\n" +
+                "   <run-up symbol=\"{$symbol}\">\n" +
+                "      <start-date>{fn:data($first/date)}</start-date>\n" +
+                "      <start-price>{fn:data($first/price)}</start-price>\n" +
+                "      <end-date>{fn:data($last/date)}</end-date>\n" +
+                "      <end-price>{fn:data($last/price)}</end-price>\n" +
+                "   </run-up>";
+
+        // parse the query into the internal syntax tree
+        final XQueryContext context = new XQueryContext();
+        context.importModule(Function.BUILTIN_FUNCTION_NS, "fn", new AnyURIValue[] { new AnyURIValue("java:" + FnModule.class.getName()) });
+        final XQueryLexer lexer = new XQueryLexer(context, new StringReader(query));
+        final XQueryParser xparser = new XQueryParser(lexer);
+        xparser.xpath();
+        if (xparser.foundErrors()) {
+            fail(xparser.getErrorMessage());
+        }
+
+        final XQueryAST ast = (XQueryAST) xparser.getAST();
+
+        final XQueryTreeParser treeParser = new XQueryTreeParser(context);
+        final PathExpr expr = new PathExpr(context);
+        treeParser.xpath(ast, expr);
+        if (treeParser.foundErrors()) {
+            fail(treeParser.getErrorMessage());
+        }
+
+        assertTrue(expr.getFirst() instanceof WindowExpr, "Expression should be of type WindowExpr");
+        assertEquals(WindowExpr.WindowType.TUMBLING_WINDOW, ((WindowExpr) expr.getFirst()).getWindowType());
+        assertEquals(new QName("first"), ((WindowExpr) expr.getFirst()).getWindowStartCondition().getCurrentItem());
+        assertEquals(null, ((WindowExpr) expr.getFirst()).getWindowStartCondition().getPosVar());
+        assertEquals(null, ((WindowExpr) expr.getFirst()).getWindowStartCondition().getPreviousItem());
+        assertEquals(new QName("second"), ((WindowExpr) expr.getFirst()).getWindowStartCondition().getNextItem());
+        assertEquals(new QName("last"), ((WindowExpr) expr.getFirst()).getWindowEndCondition().getCurrentItem());
+        assertEquals(null, ((WindowExpr) expr.getFirst()).getWindowEndCondition().getPosVar());
+        assertEquals(null, ((WindowExpr) expr.getFirst()).getWindowEndCondition().getPreviousItem());
+        assertEquals(new QName("beyond"), ((WindowExpr) expr.getFirst()).getWindowEndCondition().getNextItem());
+        assertEquals(false, ((WindowExpr) expr.getFirst()).getWindowEndCondition().isOnly());
```

Comment:
Any special reason not using `assertFalse` rsp. `assertNull` for the boolean and null cases?